### PR TITLE
fix problem with Solaar not applying saved settings for features discovered at run time

### LINKED ISSUE #551: Solaar doesn't sometime load configs
Hi,
I have problem with my Logitech MX Master 2S mouse and Solaar.
Seems Solaar doesn't load config sometime when I boot my computer. This happen about 1/4 times when I boot.
OS is 64-bit Ubuntu 18.04 Mate edition.

Edit: Looks like it load configs if I wakeup mouse before Ubuntu boot.
If I don't wakeup mouse, I need quit solaar and relaunch it.

### PR DESCRIPTION
**Pull request by pfps**
Solaar attaches the saved device settings to each individual setting, but this is not done for features discovered at run time.  This patch moves the saved device settings from the individual settings to the device itself, which is, I think, a better place for them.  Individual settings then get the saved device settings from the device and also store changes there.
Fixes #551

### CODE REVIEW

**Comment by pfps** [comment]
@lukasmichel has volunteered to test out the changes.

**Comment by pfps** [comment]
Fixes #642

**Comment by pfps** [comment]
Probably fixes #467

**Comment by lukasmichel** [comment]
Answering your instructions from issue #551 
I have tested with MX Master 2S and MX Anywhere 2. MX Anywhere lacks smart shift, so it is expected to only see 3 settings in solaar.

> Check that when you start with the mouse off you see two settings. 

Yes for the Master 2S. When I start solaar with the mouse off, I see 2 settings (High Resolution Wheel Invert and Wheel resolution) with correct values. For the MX Anywhere, I see no settings.

> When you turn the mouse on you should see four settings and the new settings should have their saved values. 

Works for both mice. For the Master 2S, I see 4 settings with the correct values. For the Anywhere 2, I see 3 settings with correct values

> You should be able to change all four settings and the changes should come back even when you stop and start solaar and turn the mouse off and on.

This works. Changed settings are loaded correctly also when turning the mice off and on and when restarting solaar.

In summary, this seems to fix the problems for the MX Master 2S and the MX Anywhere 2.

**Comment by pfps** [comment]
Fixes #463

File: lib/logitech_receiver/settings.py [hunk 1]
```diff
@@ -122,15 +124,15 @@ def write(self, value):
 		assert value is not None
 
 		if _log.isEnabledFor(_DEBUG):
-			_log.debug("%s: write %r to %s", self.name, value, self._device)
+			_log.debug("%s: Settings write %r to %s", self.name, value, self._device)
```
Comment (FFY00) [inline lib/logitech_receiver/settings.py]: nitpick: lowercase (`settings`)

File: lib/logitech_receiver/settings.py [hunk 2]
```diff
@@ -140,7 +142,7 @@ def write(self, value):
 			data_bytes = self._validator.prepare_write(value, current_value)
 			if data_bytes is not None:
 				if _log.isEnabledFor(_DEBUG):
-					_log.debug("%s: prepare write(%s) => %r", self.name, value, data_bytes)
+					_log.debug("%s: Settings prepare write(%s) => %r", self.name, value, data_bytes)
```
Comment (FFY00) [inline lib/logitech_receiver/settings.py]: nitpick: lowercase (`settings`)

**Comment by FFY00** [comment]
With which devices was this tested?

**Comment by pfps** [comment]
I did M185n (with pointer_speed removed from descriptors.py) and M510 and M310 (both of which unaffected by bug).
lukasmichel did MX Master 2S and MX Anywhere 2

File: lib/logitech_receiver/settings.py (hunk 2 above)
Comment (pfps) [reply lib/logitech_receiver/settings.py]: I force-pushed a commit with the 3 Settings turned into settings.
